File: Part 2 - Regression/Polynomial Regression/polynomial_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sun Mar 29  2020

@author: Swapnil Vivek Kulkarni
"""
#Importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Import dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
Y = dataset.iloc[:,2].values

#Fitting Linear regression in to dataset 
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X, Y)

#Fitting Polynimial regression in to dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures()
X_poly = poly_reg.fit_transform(X)
lin_reg_2 = LinearRegression()
lin_reg_2.fit(X_poly, Y)

#Visualizing Linear regression results
plt.scatter(X, Y, color = 'red')
plt.plot(X, lin_reg.predict(X),color='blue')
plt.title('Truth or bluff(Linear regression)')
plt.xlabel('Position level')
plt.xlabel('Salary')
plt.show()

#Visualizing polynomial regression results 
plt.scatter(X, Y, color = 'red')
plt.plot(X, lin_reg_2.predict(poly_reg.fit_transform(X)),color='blue')
plt.title('Truth or bluff(Linear regression)')
plt.xlabel('Position level')
plt.xlabel('Salary')
plt.show()

#predict new result with Linear Regression
lin_reg.predict(np.array([6.5]).reshape(1, 1))
#predict new result with Polynomial Regression
lin_reg_2.predict(poly_reg.fit_transform(np.array([6.5]).reshape(1, 1)))

File: Part 2 - Regression/Polynomial Regression/polynomial_regression.R
#Polynomial Regression

#@author: Swapnil Vivek Kulkarni

#importing dataset 
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

#Fitting linear regression model to dataset
lin_reg = lm(formula = Salary ~ .,
             data=dataset)

#Fitting Polynomial regression model to dataset
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
poly_reg = lm(formula = Salary ~ .,
              data= dataset) #Salary ~ . means Salary vs all other variables 

#Visualizing Liniear regression results 
#install.packages('ggplot2')
library(ggplot2)

ggplot()+
  geom_point(aes(x = dataset$Level, y =dataset$Salary),
             colour = 'red')+
  geom_line(aes(x=dataset$Level, y=predict(lin_reg,newdata = dataset)),
            colour ='blue')+
  ggtitle('Truth vs Bluff(Linear Regression)')+
  xlab('Level')+
  ylab('Salary')

#Visualizing Polynomial regression results
ggplot()+
  geom_point(aes(x = dataset$Level, y =dataset$Salary),
             colour = 'red')+
  geom_line(aes(x=dataset$Level, y=predict(poly_reg,newdata = dataset)),
            colour ='blue')+
  ggtitle('Truth vs Bluff(Polynomial Regression)')+
  xlab('Level')+
  ylab('Salary')

#Predicting new result with Linear regression
y_pred = predict(lin_reg, data.frame(Level = 6.5))

#Predicting new result with Polynomial regression
y_pred = predict(poly_reg, data.frame(Level = 6.5,
                                     Level2 = 6.5^2,
                                     Level3 = 6.5^3,
                                     Level4 = 6.5^4))

